paramak/shape.py

```diff
@@ -96,6 +100,27 @@ def __init__(
         # self.volume = None
         self.hash_value = None
 
+    @property
+    def scaling_coefficient(self):
+        return {'mm': 1, 'cm': 10, 'm': 1000}[self.units]
+
+    @scaling_coefficient.setter
+    def scaling_coefficient(self, value):
+        self._scaling_coefficient = value
+
+    @property
+    def scaled_points(self):
+        scaled_points = []
+        for p in self.points:
+            scaled_coordinate = (p[0] * self.scaling_coefficient,
+                                 p[1] * self.scaling_coefficient)
+            scaled_points.append(scaled_coordinate)
+        return scaled_points
```

Comment:
I think the potential connection type (ie straight, spline, circle) is being removed in the scaled_points here

Reply:
ah yes it is not being added to scaled_points. But it is still in points and the Shape class finds the connection type in points instead of scaled points.